src/main/app/src/constants/ActivitiesFilterBar.js

```diff
@@ -1,5 +1,98 @@
 import React from 'react';
+import filterActivities from '../hooks/browseActivitiesFilter.js';
 
+const ActivitiesFilterBar = ({activityType, activityTypes, links, updateFilteredLinks}) => {
+  // Filter states
+  const [numOfPLayers, updateNumOfPlayers] = React.useState("");
+  const [videoType, updateVideoType] = React.useState("*");
+  const [articleType, updateArticleType] = React.useState("*");
+  const [articleLength, updateArticleLength] = React.useState("*");
+
+  /*
+    This is the function that will be called when the filter button is clicked.
+    This function calls the function that is declared in hooks folder to filter out games based on the entered value.
+  */
+  const filterButtonClick = evt => {
+    const linkFilters = {}
+    
+    switch(activityType) {
+      case activityTypes.GAMES:
+        linkFilters.numOfPLayers = numOfPLayers;
+        break;
+      case activityTypes.VIDEOS:
+        linkFilters.videoType = videoType;
+        break;
+      case activityTypes.ARTICLES:
+        linkFilters.articleType = articleType;
+        linkFilters.articleLength = articleLength;
+        break;
+      default:
+        console.log("Error: Invalid activity type selection.");
+    }
+
+    filterActivities(links, activityType, activityTypes, linkFilters, updateFilteredLinks);
+  }
+
+  // When reset button is clicked, all the state go back to its default value (i.e, no filter) So, it displays all the links.
+  const filterResetClick = evt => {
+    switch(activityType) {
+      case activityTypes.GAMES:
+        updateNumOfPlayers("");
+        break;
+      case activityTypes.VIDEOS:
+        updateVideoType("*");
+        break;
+      case activityTypes.ARTICLES:
+        updateArticleType("*");
+        updateArticleLength("*");
+        break;
+      default:
+        console.log("Error: Invalid activity type selection.");
+    }
+    updateFilteredLinks(links);
+  }
+
+  /** This is the function that gets triggered on typing any value in the filter textfield.
+      It sets the filters differently depending on the activity type selected. */
+  const handleFilterChange = evt => {
+    const value = evt.target.value;
+    const name = evt.target.name;
+
+    switch(activityType) {
+      case activityTypes.GAMES:
+        updateNumOfPlayers(value);
+        break;
+      case activityTypes.VIDEOS:
+        updateVideoType(value);
+        break;
+      case activityTypes.ARTICLES:
+        if (name === "ArticleType")
+          updateArticleType(value);
+        else
```

Comment:
nit: should this be else if (name === "ArticleLength")?

Also maybe add another console.log error in an else statement

Reply:
Yeah, I think that would be best as well.

Reply:
Fixed it